allennlp/modules/stacked_alternating_lstm.py

```diff
@@ -50,8 +56,43 @@ def __init__(self,
             layers.append(layer)
         self.lstm_layers = layers
 
-    def forward(self, inputs: PackedSequence):  # pylint: disable=arguments-differ
+    def forward(self,  # pylint: disable=arguments-differ
+                inputs: PackedSequence,
+                hidden_state: torch.Tensor = None):
+        """
+        Parameters
+        ----------
+        inputs : ``PackedSequence``, required.
+            A batch first ``PackedSequence`` to run the stacked LSTM over.
+        hidden_state : ``torch.Tensor``, optional, (default = None).
+            A tensor of shape (num_layers, batch_size, hidden_size) to be used as the
+            initial hidden state for the respective layers.
+
+        Returns
+        -------
+        current : PackedSequence
+            The encoded sequence of shape (batch_size, sequence_length, hidden_size)
+        final_states: torch.Tensor
+            The per-layer final (state, memory) states of the LSTM, each with shape
+            (num_layers, batch_size, hidden_size).
+        """
+        if not hidden_state:
+            hidden_states = [None] * len(self.lstm_layers)
+        elif hidden_state.size()[0] != len(self.lstm_layers):
+            raise ConfigurationError("Initial states were passed to forward() but the number of "
+                                     "initial states does not match the number of layers.")
+        else:
+            hidden_states = hidden_state.split(1, 0)
+
         current = inputs
-        for layer in self.lstm_layers:
-            current = layer(current)
-        return current
+        final_states = []
+        for layer, state in zip(self.lstm_layers, hidden_states):
+            # The state is duplicated to mirror the Pytorch API for LSTMs.
```

Comment:
Ah, I miss read some conditionals in the pytorch rnn code which made me think that it internally duplicates the state if you pass it. I'll change this, thanks for double checking.